config/tests/jobs/jobs_test.go

```diff
@@ -979,12 +977,78 @@ func isPodQOSGuaranteed(spec *coreapi.PodSpec) bool {
 	for _, c := range spec.Containers {
 		for _, r := range resourceNames {
 			limit, ok := c.Resources.Limits[r]
-			if !ok || limit.Cmp(zero) == 0 || limit.Cmp(c.Resources.Requests[r]) != 0 {
-				isGuaranteed = false
+			if !ok {
+				errs = append(errs, fmt.Errorf("container '%v' should have resources.limits[%v] specified", c.Name, r))
+			}
+			request, ok := c.Resources.Requests[r]
+			if !ok {
+				errs = append(errs, fmt.Errorf("container '%v' should have resources.requests[%v] specified", c.Name, r))
 			}
+			if limit.Cmp(zero) == 0 {
+				errs = append(errs, fmt.Errorf("container '%v' resources.limits[%v] should be non-zero", c.Name, r))
+			} else if limit.Cmp(request) != 0 {
+				errs = append(errs, fmt.Errorf("container '%v' resources.limits[%v] (%v) should match request (%v)", c.Name, r, limit.String(), request.String()))
+			}
+		}
+	}
+	return errs
+}
+
+// isPodQOSGuaranteed returns true if the PodSpec's containers have non-zero
+// resource limits that are equal to their resource requests
+func isPodQOSGuaranteed(spec *coreapi.PodSpec) bool {
+	return len(verifyPodQOSGuaranteed(spec)) == 0
+}
+
+// A job is merge-blocking if it:
+// - is not optional
+// - reports (aka does not skip reporting)
+// - always runs OR runs if some path changed
+func isMergeBlocking(job cfg.Presubmit) bool {
+	return !job.Optional && !job.SkipReport && (job.AlwaysRun || job.RunIfChanged != "")
+}
+
+func isKubernetesReleaseBlocking(job cfg.JobBase) bool {
+	re := regexp.MustCompile(`sig-release-(1.[0-9]{2}|master)-blocking`)
+	dashboards, ok := job.Annotations["testgrid-dashboards"]
+	if !ok {
+		return false
+	}
+	return re.MatchString(dashboards)
+}
+
+// TODO: s/Should/Must and s/Logf/Errorf when all jobs pass
+func TestKubernetesMergeBlockingJobsShouldHavePodQOSGuaranteed(t *testing.T) {
+	repo := "kubernetes/kubernetes"
+	jobs := c.AllStaticPresubmits([]string{repo})
```

Comment:
Is this not also captured in `allStaticJobs?`

Reply:
`JobBase` doesn't provide info about what repo that job is for if it's a presubmit or a postubmit, so if I use `allStaticJobs` I have no way of filtering for jobs that run against "kubernetes/kubernetes"